src/meta/src/hummock/manager/compaction_group_manager.rs

```diff
@@ -617,67 +636,46 @@ impl HummockManager {
 
         // We've already lowered the default limit for write limit in PR-12183, and to prevent older clusters from continuing to use the outdated configuration, we've introduced a new logic to rewrite it in a uniform way.
         let mut rewrite_cg_ids = vec![];
-        let mut restore_cg_to_partition_vnode: HashMap<u64, BTreeMap<u32, u32>> =
-            HashMap::default();
-        for (cg_id, compaction_group_config) in &mut configs {
+        for group in compaction_groups {
             // update write limit
+            let cg_id = group.group_id;
             let relaxed_default_write_stop_level_count = 1000;
-            if compaction_group_config
-                .compaction_config
-                .level0_sub_level_compact_level_count
+            if group.compaction_config.level0_sub_level_compact_level_count
                 == relaxed_default_write_stop_level_count
             {
-                rewrite_cg_ids.push(*cg_id);
-            }
-
-            if let Some(levels) = current_version.levels.get(cg_id) {
-                if levels.member_table_ids.len() == 1 {
-                    restore_cg_to_partition_vnode.insert(
-                        *cg_id,
-                        vec![(
-                            levels.member_table_ids[0],
-                            compaction_group_config
-                                .compaction_config
-                                .split_weight_by_vnode,
-                        )]
-                        .into_iter()
-                        .collect(),
-                    );
-                }
+                rewrite_cg_ids.push(cg_id);
             }
         }
 
-        if !rewrite_cg_ids.is_empty() {
-            tracing::info!("Compaction group {:?} configs rewrite ", rewrite_cg_ids);
+        tracing::info!("Compaction group {:?} configs rewrite ", rewrite_cg_ids);
 
-            // update meta store
-            let result = compaction_group_manager
-                .update_compaction_config(
+        let update_configs =  compaction_groups_txn.update_compaction_config(
                     &rewrite_cg_ids,
                     &[
                         MutableConfig::Level0StopWriteThresholdSubLevelNumber(
                             risingwave_common::config::default::compaction_config::level0_stop_write_threshold_sub_level_number(),
                         ),
                     ],
-                )
-                .await?;
+                )?;
 
-            // update memory
-            for new_config in result {
-                configs.insert(new_config.group_id(), new_config);
-            }
-        }
-
-        compaction_group_manager.write_limit =
-            calc_new_write_limits(configs, HashMap::new(), &versioning_guard.current_version);
+        commit_multi_var!(self.meta_store_ref(), compaction_groups_txn)?;
+        compaction_group_manager.write_limit = calc_new_write_limits(
```

Comment:
Do we always update the `write_limit` every time after we update the compaction group? If so, we can introduce a real `CompactionGroupManagerTransaction`, which not only apply the change on compaction group btreemap, but also update the write_limit in the txn.

Reply:
This is not strongly related, and is executed during init. The guarantee of transaction is not needed for the time being, and consensus can be reached eventually.